proposals/20230530-driver-kernel-testing-framework.md

```diff
@@ -0,0 +1,454 @@
+                <td>ArchLinux</td>
+            </tr>
+            <tr>
+                <td>BottleRocket</td>
+            </tr>
+            <tr>
+                <td>CentOS</td>
+            </tr>
+            <tr>
+                <td>Debian</td>
+            </tr>
+            <tr>
+                <td>Fedora</td>
+            </tr>
+            <tr>
+                <td>Flatcar</td>
+            </tr>
+            <tr>
+                <td>Minikube</td>
+            </tr>
+            <tr>
+                <td>OpenSUSE</td>
+            </tr>
+            <tr>
+                <td>OracleLinux</td>
+            </tr>
+            <tr>
+                <td>PhotonOS</td>
+            </tr>
+            <tr>
+                <td>RockyLinux</td>
+            </tr>
+            <tr>
+                <td>Talos</td>
+            </tr>
+            <tr>
+                <td>Ubuntu</td>
+            </tr>
+        </table>
+</details> 
+
+</br>
+
+
+*Kernel Versions*
+
+To achieve adequate and realistic coverage, the statistical sampling of versions across distributions will prioritize the minimum and maximum supported kernel versions per driver type. Additionally, a particular emphasis will be placed on selecting LTS (Long-Term Support) releases when choosing the remaining kernel versions. As part of this approach, a minimum of 10 kernel versions will be carefully selected for each driver type.
+
+*Architectures*
+
+
+Cover each officially supported architecture by The Falco Project.
+
+The prioritization of officially supported architectures for Falco is based on their adoption levels, with `x86_64` considered as P0 (highest priority) and `aarch64` as P1 (next priority level). For any other architectures supported by its libraries, a best effort approach (lower priority) is taken (e.g. `s390x`).
+
+
+*Driver Type*
+
+Ensure equal testing coverage for each driver, taking into account the different minimum kernel versions they support.
+
+*Compiler Versions*
+
+Select the most appropriate compiler version and build container for the CI-integrated tests. Apart from the compiler version, the GLIBC version in the build container can also have an impact on the ability to compile the driver for a given kernel.
+
+> The expanded CI tests may necessitate the use of approximately 30 low-resource virtual machines (VMs) that run continuously 24/7. These VMs would be distributed across multiple third-party cloud providers. To adequately cover the condensed kernel test grid, it is estimated that up to 70 test runs would be required for each testing cycle. These tests can be launched using GitHub workflows leveraging SSH remote commands. The test results are then retrieved through this method as well. Initially, it would be logical to support these tests on demand only to avoid simultaneous runs that may try to access the same VM at the same time. In addition to the test VMs, it may be necessary to expand the CI workflows in terms of builder containers.
```

Comment:
We could consider to spin up VM with KVM on hosted GitHub runners, or with something like [actuated](https://github.com/orgs/community/discussions/8305#discussioncomment-5011953) (on microVMs).